pkg/providers/aws/provider_redis.go

```diff
@@ -255,26 +255,30 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 		}
 		return nil, "started elasticache provision", nil
 	}
-
-	// check elasticache phase
-	if *foundCache.Status != "available" {
-		logger.Infof("found instance %s current status %s", *foundCache.ReplicationGroupId, *foundCache.Status)
-		return nil, croType.StatusMessage(fmt.Sprintf("createReplicationGroup() in progress, current aws elasticache status is %s", *foundCache.Status)), nil
-	}
-	logger.Infof("found existing elasticache cluster %s", *foundCache.ReplicationGroupId)
-
 	cacheClustersOutput, err := cacheSvc.DescribeCacheClusters(&elasticache.DescribeCacheClustersInput{})
 	if err != nil {
 		errMsg := "failed to describe clusters"
 		return nil, croType.StatusMessage(fmt.Sprintf(errMsg)), errorUtil.Wrapf(err, errMsg)
 	}
 	var replicationGroupClusters []elasticache.CacheCluster
 	for _, checkedCluster := range cacheClustersOutput.CacheClusters {
-		if *checkedCluster.ReplicationGroupId == *foundCache.ReplicationGroupId {
-			replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
+		if *checkedCluster.ReplicationGroupId != "" && *foundCache.ReplicationGroupId != "" {
+			if *checkedCluster.ReplicationGroupId == *foundCache.ReplicationGroupId {
+				replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
+				if checkedCluster.EngineVersion != nil && r.Status.Version != *checkedCluster.EngineVersion {
+					r.Status.Version = *checkedCluster.EngineVersion
+				}
+			}
 		}
 	}
 
+	// check elasticache phase
+	if *foundCache.Status != "available" {
+		logger.Infof("found instance %s current status %s", *foundCache.ReplicationGroupId, *foundCache.Status)
+		return nil, croType.StatusMessage(fmt.Sprintf("createReplicationGroup() in progress, current aws elasticache status is %s", *foundCache.Status)), nil
+	}
+	logger.Infof("found existing elasticache cluster %s", *foundCache.ReplicationGroupId)
```

Comment:
I think this log message can be moved back. It makes sense to log that the cache has been found after we check that `foundCache` is not equal to nil

Reply:
:+1: